pkg/build/api/v1/types.go

```diff
@@ -112,6 +115,41 @@ const (
 	BuildPhaseCancelled BuildPhase = "Cancelled"
 )
 
+// StatusReason is a brief CamelCase string that describes a temporary or
+// permanent build error condition, meant for machine parsing and tidy display
+// in the CLI.
+type StatusReason string
+
+// These are the valid reasons of build statuses.
+const (
+	// StatusReasonError is a generic reason for a build error condition.
+	StatusReasonError StatusReason = "Error"
+
+	// StatusReasonCannotCreateBuildPodSpec is an error condition when the build
+	// strategy cannot create a build pod spec.
+	StatusReasonCannotCreateBuildPodSpec = "CannotCreateBuildPodSpec"
+
+	// StatusReasonCannotCreateBuildPod is an error condition when a build pod
+	// cannot be created.
+	StatusReasonCannotCreateBuildPod = "CannotCreateBuildPod"
+
+	// StatusReasonInvalidOutputReference is an error condition when the build
+	// output is an invalid reference.
+	StatusReasonInvalidOutputReference = "InvalidOutputReference"
+
+	// StatusReasonCancelBuildFailed is an error condition when cancelling a build
+	// fails.
+	StatusReasonCancelBuildFailed = "CancelBuildFailed"
+
+	// StatusReasonBuildPodDeleted is an error condition when the build pod is
+	// deleted before build completion.
+	StatusReasonBuildPodDeleted = "BuildPodDeleted"
+
+	// StatusReasonExceededRetryTimeout is an error condition when the build has
+	// not completed and retrying the build times out.
+	StatusReasonExceededRetryTimeout = "ExceededRetryTimeout"
+)
```

Comment:
(03:36:56 PM) claytonc: bparees: i'm just going to hazard a guess, but if you're talking about constants, we should only put constants in internal unless they're different
(03:36:59 PM) claytonc: and since constants shouldn't change
(03:37:06 PM) claytonc: we should always just add new constants to the internal api
(03:37:13 PM) claytonc: regarding the structs
(03:37:20 PM) claytonc: you should add them to the v1beta3 structs and convertsions
(03:37:31 PM) bparees: claytonc: yeah i was only talking about the constants.